### ISSUE TITLE
GetNativeVariantForObject tests are missing cases

### ISSUE THREAD

**Issue by tmat** [dotnet/coreclr#5424]
https://github.com/dotnet/coreclr/blob/a85c2686820fa9f79dc5b40e975dcb0e59367a52/tests/src/Interop/MarshalAPI/GetNativeVariantForObject/GetNativeVariantForObject.cs is missing some test cases (actually I'd say it's missing a lot of test cases). At least marshalling of strings that contain NUL characters should be explicitly tested.

**Comment by tijoytom** [comment]
@tmat Did you see any specific issue with string that contain '\0' ?

**Comment by tmat** [comment]
No, I was just checking how GetNativeVariantForObject handles it and found these tests that were missing use cases I would expect to find.

**Comment by danmosemsft** [comment]
@yizhang82 do we need this to ship 2.0? If not can you please move to Future.

**Comment by yizhang82** [comment]
Not necessary for 2.0
